### PR TITLE
Use -Command flag on Nano

### PR DESCRIPTION
**Pull request by mcquin**
Nano only supports the `-Command` flag.

### CODE REVIEW

File: spec/functional/resource/powershell_script_spec.rb [hunk 1]
```diff
@@ -287,6 +314,8 @@
     end
 
     it "executes a script with a 32-bit process if :i386 arch is specified" do
+      pending "executing scripts with a 32-bit process should raise an error on nano" if Chef::Platform.windows_nano_server?
```
Comment (mcquin) [inline spec/functional/resource/powershell_script_spec.rb]: Fixing this is outside the scope of this PR. It will be addressed separately.

File: spec/functional/resource/powershell_script_spec.rb [hunk 2]
```diff
@@ -114,24 +120,33 @@
...
     it "raises a Mixlib::ShellOut::ShellCommandFailed error if the script is not syntactically correct" do
+      pending "powershell.exe always exits with 0 on nano" if Chef::Platform.windows_nano_server?
+
       resource.code('if({)')
       resource.returns(0)
       expect { resource.run_action(:run) }.to raise_error(Mixlib::ShellOut::ShellCommandFailed)
     end
 
     it "raises an error if the script is not syntactically correct even if returns is set to 1 which is what powershell.exe returns for syntactically invalid scripts" do
+      # TODO: Why does this test succeed when the above test fails?
```
Comment (mcquin) [inline spec/functional/resource/powershell_script_spec.rb]: It's not clear to me, yet, what is happening here.

File: spec/functional/resource/powershell_script_spec.rb (hunk 1 above)
Comment (mcquin) [inline spec/functional/resource/powershell_script_spec.rb]: Note: check out this method ` node_supports_windows_architecture?`

File: spec/support/shared/functional/windows_script.rb [hunk 3]
```diff
@@ -52,6 +54,8 @@
   shared_examples_for "a script resource with architecture attribute" do
     context "with the given architecture attribute value" do
       let(:resource_architecture) { architecture }
+      # TODO: Why does expected_architecture default to :i386 and not
+      # the machine architecture?
```
Comment (mcquin) [inline spec/support/shared/functional/windows_script.rb]: I don't understand this default. The WindowsScript provider (which is the parent class of PowershellScript) is initialized with

```ruby
      def initialize( new_resource, run_context, script_extension='')
        super( new_resource, run_context )
        @script_extension = script_extension

        target_architecture = new_resource.architecture.nil? ?
          node_windows_architecture(run_context.node) : new_resource.architecture

        @is_wow64 = wow64_architecture_override_required?(run_context.node, target_architecture)

        @is_forced_32bit = forced_32bit_override_required?(run_context.node, target_architecture)
      end
```

and `node_windows_architecture` is given as

```ruby
      def node_windows_architecture(node)
        node[:kernel][:machine].to_sym
      end
```

It seems like the default should be the machine architecture and not `:i386`.

Comment (chefsalim) [inline spec/support/shared/functional/windows_script.rb]: Sounds good to me :)

Comment (mcquin) [inline spec/support/shared/functional/windows_script.rb]: Wow the stubs in the test were not doing the right thing. It was always testing i386 resource architecture and guard architecture.

**Comment by mcquin** [comment]
Ready for review @chef/client-windows

File: lib/chef/provider/powershell_script.rb [hunk 4]
```diff
@@ -107,6 +111,8 @@ def validate_script_syntax!
       end
 
       def default_interpreter_flags
+        return [] if Chef::Platform.windows_nano_server?
```
Comment (jaym) [inline lib/chef/provider/powershell_script.rb]: Does nano support `-NoProfile` and `-ExecutionPolicy`

Comment (mcquin) [inline lib/chef/provider/powershell_script.rb]: screen shot 2015-10-01 at 15 45 55

File: spec/functional/resource/powershell_script_spec.rb [hunk 5]
```diff
@@ -508,6 +564,8 @@
       end
 
       it "evaluates a 32-bit resource with a 32-bit guard and interprets boolean false as zero status code using convert_boolean_return for not_if" do
+        skip "executing scripts with a 32-bit process should raise an error on nano" if Chef::Platform.windows_nano_server?
```
Comment (jaym) [inline spec/functional/resource/powershell_script_spec.rb]: can `:not_supported_on_nano` be used here?

**Comment by jaym** [comment]
:+1:

**Comment by mwrock** [comment]
:+1:

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## Non-GitHub Verified Committers

There are 1 commit author(s) whose commits are authored by a non-GitHub verified email address. Chef will have to manually verify that they are authorized to contribute.

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)
